generators/kubernetes/templates/deployment.yml.ejs

```diff
@@ -76,12 +81,26 @@ spec:
               key: registry-admin-password
         - name: EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE
           value: http://admin:${jhipster.registry.password}@jhipster-registry.<%= kubernetesNamespace %>.svc.cluster.local:8761/eureka/
+          <%_ if (app.istioSupportOpt !== 'no') { _%>
+        - name: EUREKA_INSTANCE_PREFER_IP_ADDRESS
+          value: "false"
+        - name: EUREKA_INSTANCE_HOSTNAME
+          value: <%= app.baseName.toLowerCase() %>
+          <%_ } _%>
         <%_ } _%>
         <%_ if (app.serviceDiscoveryType === 'consul') { _%>
         - name: SPRING_CLOUD_CONSUL_HOST
           value: consul.<%= kubernetesNamespace %>.svc.cluster.local
         - name: SPRING_CLOUD_CONSUL_PORT
           value: "8500"
+          <%_ if (app.istioSupportOpt !== 'no') { _%>
+        - name: SPRING_CLOUD_CONSUL_DISCOVERY_PREFER_IP_ADDRESS
+          value: "false"
+        - name: SPRING_CLOUD_CONSUL_DISCOVERY_HOSTNAME
+          value: <%= app.baseName.toLowerCase() %>
+        - name: SPRING_CLOUD_CONSUL_DISCOVERY_SERVICE_NAME
+          value: <%= app.baseName.toLowerCase() %>
+          <%_ } _%>
```

Comment:
can we add one more conditional env var to propagate additional Istio headers? e.g.:

```
- name: SPRING_SLEUTH_PROPAGATION_KEYS
   value: "x-request-id,x-ot-span-context"
```

Reply:
Sure, I guess sleuth is not available as a dependency. Have to add it to the build file first.

Reply:
Sleuth is available when packaging the app with the zipkin maven profile.

Reply:
@saturnism done!

Reply:
nice!